connector/spanmetricsconnector/connector.go

```diff
@@ -233,41 +232,109 @@ func (p *connectorImp) buildMetrics() pmetric.Metrics {
 	ilm := m.ResourceMetrics().AppendEmpty().ScopeMetrics().AppendEmpty()
 	ilm.Scope().SetName("spanmetricsconnector")
 
-	p.buildCallsSumMetrics(ilm)
-	p.buildLatencyHistogramMetrics(ilm)
+	p.buildCallsMetrics(ilm)
+
+	switch histograms := p.histograms.(type) {
```

Comment:
Instead of type switching on concrete type, is it possible to move the ownership of `buildLatencyMetricsFromExplicitHistograms` and `buildLatencyMetricsFromExponentialHistograms` into `ExplicitHistogramMetrics` and `ExponentialHistogramMetrics` respectively?

It means that this can be simplified with something like `p.histograms.buildLatencyMetrics(...)` and let the runtime types own its own specific knowledge of how to build latency metrics.

Reply:
Sounds good, I'll add that.